File: remote-php/tests/HealthResponseTest.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/../src/HealthResponse.php';

use DriveAura\Remote\HealthResponse;

$expected = ['apiVersion' => '1.0', 'service' => 'remote-php', 'status' => 'ok'];
if (HealthResponse::body() !== $expected) {
    fwrite(STDERR, "Contratto salute PHP non valido.\n");
    exit(1);
}

echo "Contratto salute PHP valido.\n";

File: bridge-servlet/core/src/test/java/it/unibg/driveaura/bridge/core/HealthResponseTest.java
package it.unibg.driveaura.bridge.core;

/** Dependency-free automated contract test runnable with javac and java. */
public final class HealthResponseTest {
    public void testHealthContract() {
        main(new String[0]);
    }

    public static void main(String[] args) {
        String expected = "{\"apiVersion\":\"1.0\",\"service\":\"bridge-servlet\",\"status\":\"ok\"}";
        if (!expected.equals(HealthResponse.json())) {
            throw new AssertionError("Contratto salute bridge non valido.");
        }
        System.out.println("Contratto salute bridge valido.");
    }
}
